mithril-client-cli/src/commands/tools/snapshot_converter.rs

```diff
@@ -0,0 +1,963 @@
+                Err(_) => continue,
+            };
+
+            let path = entry.path();
+            if path.is_dir()
+                && (min_slot
+                    .as_ref()
+                    .map(|(min, _)| slot < *min)
+                    .unwrap_or(true))
+            {
+                min_slot = Some((slot, path));
+            }
+        }
+
+        min_slot.map(|(_, path)| path).ok_or_else(|| {
+            anyhow!(
+                "No valid ledger state snapshot found in directory: {}",
+                ledger_dir.display()
+            )
+        })
+    }
+
+    fn copy_oldest_ledger_state_snapshot(
+        db_dir: &Path,
+        target_dir: &Path,
+    ) -> MithrilResult<PathBuf> {
+        let snapshot_path = Self::find_oldest_ledger_state_snapshot(db_dir)?;
+        let copied_snapshot_path = copy_dir(&snapshot_path, target_dir)?;
+
+        Ok(copied_snapshot_path)
+    }
+
+    fn compute_converted_snapshot_output_path(
+        snapshots_dir: &Path,
+        input_snapshot: &Path,
+        flavor: &UTxOHDFlavor,
+    ) -> MithrilResult<PathBuf> {
+        let slot_number = Self::extract_slot_number(input_snapshot).with_context(|| {
+            format!(
+                "Failed to extract slot number from: {}",
+                input_snapshot.display()
+            )
+        })?;
+
+        let converted_snapshot_path = snapshots_dir.join(format!(
+            "{}_{}",
+            slot_number,
+            flavor.to_string().to_lowercase()
+        ));
+
+        Ok(converted_snapshot_path)
+    }
+
+    fn extract_slot_number(path: &Path) -> MithrilResult<u64> {
+        let file_name = path
+            .file_name()
+            .ok_or_else(|| anyhow!("No filename in path: {}", path.display()))?;
+
+        let file_name_str = file_name
+            .to_str()
+            .ok_or_else(|| anyhow!("Invalid UTF-8 in path filename: {:?}", file_name))?;
+
+        file_name_str
+            .parse::<u64>()
+            .with_context(|| format!("Invalid slot number in path filename: {}", file_name_str))
+    }
+
+    /// Commits the converted snapshot by replacing the current ledger state snapshots in the database directory.
+    fn commit_converted_snapshot(
+        db_dir: &Path,
+        converted_snapshot_path: &Path,
+    ) -> MithrilResult<()> {
+        let ledger_dir = db_dir.join(LEDGER_DIR);
+        println!(
+            "Upgrading and replacing ledger state in {} with converted snapshot: {}",
+            ledger_dir.display(),
+            converted_snapshot_path.display()
+        );
+
+        let filename = converted_snapshot_path
+            .file_name()
+            .ok_or_else(|| anyhow!("Missing filename in converted snapshot path"))?
+            .to_string_lossy();
+
+        let (slot_number, _) = filename
+            .split_once('_')
+            .ok_or_else(|| anyhow!("Invalid converted snapshot name format: {}", filename))?;
+
+        remove_dir_all(&ledger_dir).with_context(|| {
```

Comment:
You should delete what's inside the ledger directory but not the directory itself. This will likely be a problem when we use the command to update an existing Cardano node database.